metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/WorkflowPostProcessor.java

```diff
@@ -32,11 +47,45 @@ public WorkflowPostProcessor(DepublishedRecordDao depublishedRecordDao) {
    */
   void performPluginPostProcessing(AbstractExecutablePlugin<?> plugin, String datasetId) {
 
-    // If it is a index to publish plugin, we need to reset all the depublished record statuses.
-    if (plugin.getPluginType() == PluginType.PUBLISH) {
-      LOGGER.info("Starting postprocessing for an index to publish in dataset {}.", datasetId);
-      depublishedRecordDao.markAllAsNotDepublished(datasetId);
-      LOGGER.info("Finished postprocessing for an index to publish in dataset {}.", datasetId);
+    final PluginType pluginType = plugin.getPluginType();
+    LOGGER.info("Starting postprocessing of plugin {} in dataset {}.", pluginType, datasetId);
+    //Reset depublish status if index to PUBLISH
+    if (pluginType == PluginType.PUBLISH) {
+      publishPostProcess(datasetId);
+    } else if (pluginType == PluginType.DEPUBLISH) {
+      depublishPostProcess(plugin, datasetId);
+    }
+    LOGGER.info("Finished postprocessing of plugin {} in dataset {}.", pluginType, datasetId);
+  }
+
+  private void publishPostProcess(String datasetId) {
+    depublishedRecordDao.markRecordIdsWithDepublicationStatus(datasetId,
+        DepublicationStatus.PENDING_DEPUBLICATION, null);
+  }
+
+  private void depublishPostProcess(AbstractExecutablePlugin<?> plugin, String datasetId) {
+    final boolean datasetDepublish = ((DepublishPluginMetadata) plugin.getPluginMetadata())
+        .isDatasetDepublish();
+    if (datasetDepublish) { //Reset depublish status of records if depublishing dataset
+      depublishedRecordDao.markRecordIdsWithDepublicationStatus(datasetId,
+          DepublicationStatus.PENDING_DEPUBLICATION, null);
+      //Find latest PUBLISH Type Plugin and set dataStatus
+      final PluginWithExecutionId<MetisPlugin> latestSuccessfulPlugin = workflowExecutionDao
+          .getLatestSuccessfulPlugin(datasetId, OrchestratorService.PUBLISH_TYPES);
+      if (Objects.nonNull(latestSuccessfulPlugin) && Objects
+          .nonNull(latestSuccessfulPlugin.getPlugin())) {
+        final WorkflowExecution workflowExecutionToUpdate = workflowExecutionDao
+            .getById(latestSuccessfulPlugin.getExecutionId());
+        final Optional<AbstractMetisPlugin> metisPluginWithType = workflowExecutionToUpdate
+            .getMetisPluginWithType(latestSuccessfulPlugin.getPlugin().getPluginType());
+        if (metisPluginWithType.isPresent()) {
+          metisPluginWithType.get().setDataStatus(DataStatus.DELETED);
+          workflowExecutionDao.updateWorkflowPlugins(workflowExecutionToUpdate);
+        }
+      }
+    } else { //Set depublish status if depublishing records
+      depublishedRecordDao.markRecordIdsWithDepublicationStatus(datasetId,
+          DepublicationStatus.DEPUBLISHED, plugin.getStartedDate());
```

Comment:
This is not right. It is also not required (yet), but still. This should check the actual depublished records from the topology and only mark those. Perhaps it is best to remove this code here for now and revisit this properly during the ticket that deals with depublication of record ids?

Reply:
Okay I will remove this. Wasn't sure about it, I thought ecloud will return it as a report and not as part of the monitor call?

Reply:
We need to take care to only mark those that were in fact deleted. I think they were going to implement a service where we can obtain this information. But if we mark one as depublished and it's not, this can cause some issues.